tokio/src/runtime/mod.rs

```diff
@@ -172,6 +172,128 @@
+//! polls B. This is the case even if A yields using [`yield_now`]. All that is
+//! guaranteed is that it will poll B eventually.
+//!
+//! Normally, tasks are polled only if they have notified the runtime by waking
+//! their waker. However, this is not guaranteed, and Tokio may poll tasks that
+//! have not notified the runtime under some circumstances. This is called a
+//! spurious wakeup.
+//!
+//! ## IO and timers
+//!
+//! Beyond just polling tasks, the runtime must also manage IO resources and
+//! timers. It does this by periodically checking whether there are any IO
+//! resources or timers that are ready, and notifying the relevant task so that
+//! it will be polled.
+//!
+//! These checks are performed periodically between polling tasks. Under the
+//! same assumptions as the previous fairness guarantee, Tokio guarantees that
+//! it will notify tasks with an IO or timer event within some maximum number of
+//! time units.
+//!
+//! ## Current behavior (current thread runtime)
+//!
+//! This section describes how the [current thread runtime] behaves today. This
+//! behavior may change in future versions of Tokio.
+//!
+//! The current thread runtime maintains two FIFO queues of tasks that are ready
+//! to be polled: the global queue and the local queue. The runtime will prefer
+//! to choose the next task to poll from the local queue, and will only pick a
+//! task from the global queue if the local queue is empty, or if it has picked
+//! a task from the local queue 31 times in a row. The number 31 can be
+//! changed using the [`global_queue_interval`] setting.
+//!
+//! The runtime will check for new IO or timer events whenever there are no
+//! tasks ready to be polled, or when it has polled 61 tasks in a row. The
+//! number 61 may be changed using the [`event_interval`] setting.
+//!
+//! When a task is woken from within a task running on the runtime, then the
+//! woken task is added directly to the local queue. Otherwise, the task is
+//! added to the global queue. The current thread runtime does not use [the lifo
+//! slot optimization].
+//!
+//! ## Current behavior (multi thread runtime)
+//!
+//! This section describes how the [multi thread runtime] behaves today. This
+//! behavior may change in future versions of Tokio.
+//!
+//! A multi thread runtime has a fixed number of worker threads, which are all
+//! created on startup. The multi thread runtime maintains one global queue, and
+//! a local queue for each worker thread. The local queue of a worker thread can
+//! fit at most 256 tasks. If more than 256 tasks are added to the local queue,
+//! then half of them are moved to the global queue to make space.
+//!
+//! The runtime will prefer to choose the next task to poll from the local
+//! queue, and will only pick a task from the global queue if the local queue is
+//! empty, or if it has picked a task from the local queue 61 times in a row.
```

Comment:
In the multi-threaded case, the global queue interval is based on a heuristic using the task mean-poll time. IIRC it targets 10ms intervals and does 10ms / mean-task-poll-time to figure out the number of tasks to poll.